#### Questão 1
Macro "concatenação" de listas.

In [ ]:
(define x '(1 2 3))
(define y '(3 2 1))

(define-macro (concat a b)
  `'(,@a ,@b))

(display (concat (1 2 3) (3 2 1)))


(define-macro (concat-s a b)
  `(append ,a ,b))

(display (concat-s x y))


(define-macro (concat-d a b)
  (list append a b))

(display (concat-d x y))

(1 2 3 3 2 1)(1 2 3 3 2 1)(1 2 3 3 2 1)

#### Questão 2
for-1

In [ ]:
(define-macro (for-1 vvs . body)
  (let ([i   (car vvs)]
        [i0  (cadr vvs)]
        [il (cadr (cdr vvs))])
    `(do ((,i ,i0 (1+ ,i)))
         ((> ,i ,il))
       ,@body)))

(for-1 (x 1 5)
  (display x))

(for-1 (x 50 57)
  (display "\n")
  (display x))

12345
50
51
52
53
54
55
56
57

#### Questão 3
O que o código a seguir faz?

```scheme
(for-1 (limit 1 5)
  (display limit))
``` 

Printa os números de 1 a 5 em sequência "12345".

** se mantivesse o termo "limit" na definição do macro for-1, o programa gera erro.